addons/web_tip/static/src/js/tip.js

```diff
@@ -63,70 +69,36 @@ var Tip = Class.extend({
         var model = action.res_model;
     },
 
-    on_view: function(view) {
+    on_form_view: function(action_id, model) {
         var self = this;
-        var fields_view = view.fields_view;
-        var action_id = view.ViewManager.action ? view.ViewManager.action.id : null;
-        var model = fields_view.model;
-
-        // kanban
-        if(fields_view.type === 'kanban') {
-            var dataset_def = $.Deferred();
-            var groups_def = $.Deferred();
-            view.on("kanban_dataset_processed", self, function() {
-                var length = view.dataset.ids.length;
-                dataset_def.resolve(length);
-            });
-            view.on('kanban_groups_processed', self, function() {
-                groups_def.resolve();
-            });
-            dataset_def.done(function(length) {
-                self.eval_tip(action_id, model, fields_view.type);
-            });
-            groups_def.done(function() {
-                self.eval_tip(action_id, model, fields_view.type);
-            });
-        } else if (fields_view.type === 'tree') {
-            view.on('view_list_rendered', self, function() {
-                self.eval_tip(action_id, model, fields_view.type);
-            });
-        } else if (view.hasOwnProperty('editor')) {
-            view.on('view_list_rendered', self, function() {
-                self.reposition();
-            });
+        var type = null;
+
+        if (this.view.datarecord) {
+            type = this.view.datarecord.type ? this.view.datarecord.type : null;
```

Comment:
What would happen if you would write this on line 74:
var type = this.view.datarecord ? this.view.datarecord.type : null

If this.view.datarecord but not this.view.datarecord.type, it will result in type=undefined, is it an issue for the code involving this type variable ?